### DISCUSSION TITLE
Python virtual environment for ZOAU Python APIs

### DISCUSSION THREAD

**Discussion by gngrossi** [IBM/zoau-samples#54] [Q&A] [answered]
Having an issue activating a virtual environment.
What am I missing?
thanks

$ cd /newco/zopentools/my_env

$ python3 -m venv env

$ ls -l env
drwxr-xr-x   2 @02858   @ISCICS1    8192 Nov  9 15:20 bin
drwxr-xr-x   2 @02858   @ISCICS1       0 Nov  9 15:20 include
drwxr-xr-x   3 @02858   @ISCICS1    8192 Nov  9 15:20 lib
lrwxrwxrwx   1 @02858   @ISCICS1       3 Nov  9 15:20 lib64 -> lib
-rw-r--r--   1 @02858   @ISCICS1     114 Nov  9 15:20 pyvenv.cfg

$ cat env/pyvenv.cfg
home = /newco/zopentools/pyz/usr/lpp/IBM/cyp/v3r10/pyz/bin
include-system-site-packages = false
version = 3.10.5

$ ls -l env/bin
-rw-r--r--   1 @02858   @ISCICS1    9033 Nov  9 15:20 Activate.ps1
-rw-r--r--   1 @02858   @ISCICS1    1994 Nov  9 15:20 activate
-rw-r--r--   1 @02858   @ISCICS1     920 Nov  9 15:20 activate.csh
-rw-r--r--   1 @02858   @ISCICS1    2062 Nov  9 15:20 activate.fish
-rwxr-xr-x   1 @02858   @ISCICS1     246 Nov  9 15:20 pip
-rwxr-xr-x   1 @02858   @ISCICS1     246 Nov  9 15:20 pip3
-rwxr-xr-x   1 @02858   @ISCICS1     246 Nov  9 15:20 pip3.10
lrwxrwxrwx   1 @02858   @ISCICS1       7 Nov  9 15:20 python -> python3
lrwxrwxrwx   1 @02858   @ISCICS1      60 Nov  9 15:20 python3 -> /hewitt/zopentools/pyz/usr/lpp/IBM/cyp/v3r10/pyz/bin/python3
lrwxrwxrwx   1 @02858   @ISCICS1       7 Nov  9 15:20 python3.10 -> python3

$ source env/bin/activate
source: FSUM7351 not found

**Comment by EnsonoStevenPerva** [comment, marked as answer]
If you're not in the bash shell, you'll find that the source command is missing. The `activate` script seems to want you to use bash, for some reason (at least mine does in the top 2 lines):
```
# This file must be used with "source bin/activate" *from bash*
# you cannot run it directly
```

There is almost certain a way to get this sorted with z/OS shell, but I'd defer to someone much smarter than myself on the right way to do this...
